docs/markdown/snowplow_fractribution_macros_docs.md

```diff
@@ -0,0 +1,215 @@
+         when lower(mkt_medium) in ('cpc', 'ppc') and not regexp_count(lower(mkt_campaign), 'brand|generic') > 0 then 'Paid_Search_Other'
+         when lower(mkt_medium) = 'organic' then 'Organic_Search'
+         when lower(mkt_medium) in ('display', 'cpm', 'banner') and regexp_count(lower(mkt_campaign), 'prospect') > 0 then 'Display_Prospecting'
+         when lower(mkt_medium) in ('display', 'cpm', 'banner') and regexp_count(lower(mkt_campaign), 'retargeting|re-targeting|remarketing|re-marketing') > 0 then 'Display_Retargeting'
+         when lower(mkt_medium) in ('display', 'cpm', 'banner') and not regexp_count(lower(mkt_campaign), 'prospect|retargeting|re-targeting|remarketing|re-marketing') > 0 then 'Display_Other'
+         when regexp_count(lower(mkt_campaign), 'video|youtube') > 0 or regexp_count(lower(mkt_source), 'video|youtube') > 0 then 'Video'
+         when lower(mkt_medium) = 'social' and regexp_count(lower(mkt_campaign), 'prospect') > 0 then 'Paid_Social_Prospecting'
+         when lower(mkt_medium) = 'social' and regexp_count(lower(mkt_campaign), 'retargeting|re-targeting|remarketing|re-marketing') > 0 then 'Paid_Social_Retargeting'
+         when lower(mkt_medium) = 'social' and not regexp_count(lower(mkt_campaign), 'prospect|retargeting|re-targeting|remarketing|re-marketing') > 0 then 'Paid_Social_Other'
+         when mkt_source = '(direct)' then 'Direct'
+         when lower(mkt_medium) = 'referral' then 'Referral'
+         when lower(mkt_medium) = 'email' then 'Email'
+         when lower(mkt_medium) in ('cpc', 'ppc', 'cpv', 'cpa', 'affiliates') then 'Other_Advertising'
+         else 'Unmatched_Channel'
+    end
+```
+{% endraw %}
+{% enddocs %}
+
+{% docs macro_channel_spend %}
+
+{% raw %}
+ A macro for the user to overwrite it with a sql script to extract total ad spend by channel.
+
+#### Returns
+
+A sql script to extract channel and corresponding spend values from a data source.
+
+
+#### Usage
+
+```sql
+
+  -- Example (simplified) query:
+
+  select
+    channel,
+    sum(spend_usd) as spend
+  from example_spend_table
+  group by 1
+
+  -- Example table output for the user-supplied SQL:
+
+  Channel     |  Spend
+ ------------------------
+  direct      |  1050.02
+  paid_search |  10490.11
+  etc...
+
+```
+{% endraw %}
+{% enddocs %}
+
+
+{% docs macro_conversion_clause %}
+
+{% raw %}
+A macro to let users specify how to filter on conversion events.
+
+#### Returns
+
+A sql to be used in a WHERE clause to filter on conversion events.
+
+#### Usage
+
+```sql
+ tr_total > 0
+
+```
+{% endraw %}
+{% enddocs %}
+
+
+{% docs macro_conversion_value %}
+
+{% raw %}
+A user defined macro that specifies either a single column or a calculated value that represents the value associated with the conversion.
+
+#### Returns
+
+A sql to be used to refer to the conversion value.
+
+#### Usage
+
+```sql
+
+tr_total
+
+```
```

Comment:
This would be an example of the macro contents, it should be an example of the macro being called.